### ISSUE TITLE
support string parameters with possible values as enum

### ISSUE THREAD

**Issue by nosyjoe** [luckymarmot/Paw-SwaggerImporter#10]
swagger 2.0 allows me to specify possible parameter values (for headers, queries and so on) as an enum.

when importing, the values are not imported correctly. is this possible or does the paw api not support enum values?

example

```
"parameters": {
    "headerX": {
      "name": "headerX",
      "in": "header",
      "required": true,
      "default": "value1",
      "type": "string",
      "enum": [
          "value1",
          "value2",
          "value3"
      ]
    }
```

**Comment by letsbyteit** [comment]
@nosyjoe Just spoke with the Paw team, there is currently no UI implemented to do this. I think, you expected also kind of drop-down-menu? Yeah, me too. But it's not implemented, but they will implement it.

**Comment by nosyjoe** [comment]
@letsbyteit yeah you're right, I was expecting a kind of drop-drop down, until I figured out that Paw does import and even autocomplete the enum values - which is nice.

still, you need to start typing for the autocompletion, thus know at least the first character of the value

**Comment by hishnash** [comment]
@nosyjoe @letsbyteit I am looking into if we can find a way to let you hack this into paw in 2.3.3 through creating CustomDV objects with the importer that expose UI, therefore would have a dropdown if you click on the dv tag (not perfect but a much easier short term fix).

**Comment by nosyjoe** [comment]
@hishnash sounds awesome!

**Comment by letsbyteit** [comment]
Just a short question: How should the JSON-structure be for that? I didn't get any autocompletion so far, so I think I've to restructure my JSON?
Or does it currently just work within the parameters-array directly? I'd like to get this also for the models (in the schema-object).

@nosyjoe @hishnash

**Comment by nosyjoe** [comment]
so here's an example-excerpt from a swagger son definition, it's quite similar to the one in the initial post. When I import it (the full one, not the excerpt), I can auto-complete the enum values, when I type "de" it shows mit the two options starting with "de":

``` json
{
    "swagger": "2.0",
    ...
    "parameters": {
        "language": {
            "name": "Language",
            "in": "header",
            "required": true,
            "default": "de_DE",
            "type": "string",
            "enum": [
                "en_US",
                "de_DE",
                "de_AT"
            ]
        }
    }
}
```

I haven't tried it for models

**Comment by JonathanMontane** [comment]
fixed in v2.2.0
